src/d/actor/d_a_kita.cpp

```diff
@@ -4,52 +4,448 @@
+    mDoMtx_XrotM(*calc_mtx, i_this->mRotX);
+    mDoMtx_YrotM(*calc_mtx, -i_this->mRotY);
+    mDoMtx_XrotM(*calc_mtx, i_this->shape_angle.x);
+    mDoMtx_ZrotM(*calc_mtx, i_this->shape_angle.z);
+    i_this->mModel->setBaseTRMtx(*calc_mtx);
+
+    cXyz local48[1];
+    for(int i = 0; i < 4; i++){
+        MtxPush();
+        local48->x = i_this->scale.x * xd[i];
+        local48->y = REG0_F(5) + -15.0f;
+        local48->z = i_this->scale.z * zd[i];
+        MtxPosition(local48, &i_this->field_2E8[i]);
+        MtxPull();
+    }
+    mDoMtx_copy(*calc_mtx, i_this->mBgwMtx);
+    i_this->pm_bgw->Move();
+    if(i_this->field_360 == 2){
+        if(i_this->mBaseEmitter == NULL){
+            i_this->mBaseEmitter = dComIfGp_particle_set(0x828D, &i_this->current.pos);
+        }
+        else {
+            i_this->mBaseEmitter->setGlobalTranslation(i_this->current.pos.x, i_this->current.pos.y - (REG0_F(17) + 40.0f), i_this->current.pos.z);
+        }
+    }
+    else {
+        if(i_this->mBaseEmitter != NULL){
+            i_this->mBaseEmitter->becomeInvalidEmitter();
+            i_this->mBaseEmitter = NULL;
+        }
+    }
+
+    return TRUE;
 }
 
 /* 00001CB8-00001CC0       .text daKita_IsDelete__FP10kita_class */
 static BOOL daKita_IsDelete(kita_class*) {
-    /* Nonmatching */
+    return TRUE;
 }
 
 /* 00001CC0-00001D3C       .text daKita_Delete__FP10kita_class */
-static BOOL daKita_Delete(kita_class*) {
-    /* Nonmatching */
+static BOOL daKita_Delete(kita_class* i_this) {
+    dComIfG_resDelete(&i_this->mPhs, "Kita");
+    if(i_this->heap != NULL){
+        dComIfG_Bgsp()->Release(i_this->pm_bgw);
+    }
+    if(i_this->mBaseEmitter != NULL){
+        i_this->mBaseEmitter->becomeInvalidEmitter();
+    }
+
+    return TRUE;
 }
 
 /* 00001D3C-00001EB0       .text CallbackCreateHeap__FP10fopAc_ac_c */
-static BOOL CallbackCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CallbackCreateHeap(fopAc_ac_c* p_this) {
+    BOOL ret;
+    kita_class* actor = static_cast<kita_class*>(p_this);
+
+    J3DModelData* modelData = static_cast<J3DModelData*>(dComIfG_getObjectRes("Kita", 4));
+    actor->mModel = mDoExt_J3DModel__create(modelData, 0, 0x11020203);
+
+    if(actor->mModel == NULL){
+        ret = FALSE;
+    }
+    else {
+        JUT_ASSERT(0x3b2, modelData != 0);
+        actor->pm_bgw = new dBgW();
+        JUT_ASSERT(0x3b7, actor->pm_bgw != NULL);
+        actor->pm_bgw->Set(static_cast<cBgD_t*>(dComIfG_getObjectRes("Kita", 7)), cBgW::MOVE_BG_e, &actor->mBgwMtx);
+        actor->pm_bgw->SetCrrFunc(dBgS_MoveBGProc_Typical);
+        actor->pm_bgw->SetRideCallback(ride_call_back);
+        ret = TRUE;
+    }
+
+    return ret;
 }
 
 /* 00001EB0-00002224       .text daKita_Create__FP10fopAc_ac_c */
-static cPhs_State daKita_Create(fopAc_ac_c*) {
-    /* Nonmatching */
+static cPhs_State daKita_Create(kita_class* i_this) {
+    /* Nonmatching - regalloc and data address*/
```

Comment:
You shouldn't change function parameters. This changes the function's symbol compared to what was in the official symbol maps.

Furthermore, it's very likely that you changing this parameter is what caused the regalloc. Try doing the a_this pattern and it will probably match:
```suggestion
static cPhs_State daKita_Create(fopAc_ac_c* i_this) {
    kita_class* a_this = (kita_class*)i_this;
```